### PR TITLE
I'm on a retro spree!

### LINKED ISSUE #2337: Doors with police/engineer tape has no density [bug]
image

How to reproduce:
1. Emag door.
2. Tape door.
3. Screwdriver door.
4. Crowbar door.
5. Insert new airlock electronics into door.
6. Screwdriver door.
7. Walk through the door like harry houdini.

How to fix this bug in game:
1. Remove tape.

### PR DESCRIPTION
**Pull request by CrAzYPiLoT-SS13**
Fixes #2337. As I said, retro spree! And yes, this is my last PR of the day, relax.

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -1070,3 +1070,9 @@ About the new airlock wires panel:
 		src.open()
 		src.lock()
 	return
+
+/obj/machinery/door/airlock/proc/is_emagged() // used to keep people going through taped closed doors
```
Comment (TheDZD) [inline code/game/machinery/doors/airlock.dm]: All machines have an `emagged` var to check if they are emagged.

File: code/game/objects/items/policetape.dm [hunk 2]
```diff
@@ -125,6 +125,10 @@ var/list/tape_roll_applications = list()
 	if(istype(A, /obj/machinery/door/airlock))
 		var/turf/T = get_turf(A)
 		var/obj/item/tape/P = new tape_type(T.x,T.y,T.z)
+		var/obj/machinery/door/airlock/R = A
+		if(R.is_emagged())
+			to_chat(user, "You cannot tape this door.")
+			return // we don't want problems with density
```
Comment (TheDZD) [inline code/game/objects/items/policetape.dm]: Nothing below this even touches the door's density, I don't know how this would fix anything.

Comment (CrAzYPiLoT-SS13) [inline code/game/objects/items/policetape.dm]: The door state doesn't update when it's fixed while it's taped, that's the main problem. This just makes it so you cannot tape an emagged door.

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (CrAzYPiLoT-SS13) [inline code/game/machinery/doors/airlock.dm]: Just tested with the emagged var, it doesn't work. I guess that's why they used other methods in airlock.dm.

**Comment by CrAzYPiLoT-SS13** [comment]
A better PR has been opened.

**Pull request closed without merging**